### ISSUE TITLE
Wrong "latest version" from `docker log` when upgrading the Docker container

### ISSUE THREAD

**Issue by PeterDaveHello** [pi-hole/docker-pi-hole#1004]
Please note that this issue could be hard to reproduce, feel free to close it if the chance to happen seems too low.

### Versions

-   Pi-hole version is v5.9
-   AdminLTE version is v5.11
-   FTL version is v5.14
-   Container tag is: 2022.02.1

### Platform

- OS and version: Ubuntu 20.04 / AMD64
- Platform: Docker

### Expected behavior

From the `docker log` of Pi-Hole container, there's a section displaying the versions, the latest version expected to be the actual latest version on GitHub:

```
 ::: Docker start setup complete
  Checking if custom gravity.db is set in /etc/pihole/pihole-FTL.conf
  Pi-hole version is v5.9 (Latest: v5.9)
  AdminLTE version is v5.11 (Latest: v5.11)
  FTL version is v5.14 (Latest: v5.14)
  Container tag is: 2022.02.1
```

### Actual behavior / bug

The strange thing is I saw the latest version older than the current version, when I upgrade Pi-Hole Docker image from `2022.01.1` to `2022.02.1`:

```
 ::: Docker start setup complete
  Checking if custom gravity.db is set in /etc/pihole/pihole-FTL.conf
  Pi-hole version is v5.9 (Latest: v5.8.1)
  AdminLTE version is v5.11 (Latest: v5.10.1)
  FTL version is v5.14 (Latest: v5.13)
  Container tag is: 2022.02.1
```

### Steps to reproduce

I can't reproduce it yet.

Tried to downgrade and upgrade the Pi-Hole in the same host, same environment again, but the problem didn't appear after.

Also tried to upgrade another Pi-Hole docker instance, but the problem didn't happen.

**Comment by dschaper** [comment]
Yep, just hasn't run the 'latest' check when you first start up a new container.

**Comment by ogheinz77** [comment]
Hello,
same here, i have persisted volumes and the wrong "latest version" problem from docker appears in my constellation always.

**Comment by DFlexy** [comment]
Same problem here

![image](https://user-images.githubusercontent.com/50220928/154151414-b28043c4-c042-4a1e-af8d-756d0e584746.png)

**Comment by PromoFaux** [comment]
This is a different problem... (but one that will be fixed in the next version)

**Comment by ogheinz77** [comment]
> Hello, same here. I use persisted volumes for Pi Hole in Docker. And the wrong "latest version" problem appears in my constellation whenever I update the Pi Hole Docker.
> 
> ![Pihole-Docker](https://user-images.githubusercontent.com/99647921/153848158-383acf2d-683d-40ae-b6e9-d08e4e65bbf6.jpg)

Hello, short question. Is there a chance to fix the problem that i have reported?

**Comment by PromoFaux** [comment]
Maybe, maybe not. It's cosmetic, and usually sorts itself out after a small time, right?

I think I know where to do it, it's just getting around to it. The running of the container should not be affected, so it is lower priority

**Comment by PeterDaveHello** [comment]
@PromoFaux correct!

**Comment by ogheinz77** [comment]
> Maybe, maybe not. It's cosmetic, and usually sorts itself out after a small time, right?
> 

How sorts this itself out after a time?
In my installation it shows still the old version (Docker Tag 2021.12)

**Comment by PromoFaux** [comment]
@ogheinz77 , You're seeing something different to that which is reported in the OP, at least I think so.

Please explain exactly which part is showing the old version.

Please also provide a debug log (`pihole -t` from inside the container) and the output of `docker inspect pihole` (from outside the container - and if your container is not called `pihole`, then replace `pihole` with the name of your container)

**Comment by ogheinz77** [comment]
> @ogheinz77 , You're seeing something different to that which is reported in the OP, at least I think so.
> 
> Please explain exactly which part is showing the old version.
> 
> Please also provide a debug log (`pihole -t` from inside the container) and the output of `docker inspect pihole` (from outside the container - and if your container is not called `pihole`, then replace `pihole` with the name of your container)

@PromoFaux pihole,  Container is called pihole. I have send you a e-mail with the docker inspect pihole output.
   Hope that helps you to find the problem.

**Comment by PromoFaux** [comment]
What are you using to manage your container? You need to destroy and recreate the container between updates (per the upgrade guide in the readme) else things will start to go wrong. But it is definitely a different issue than that which was reported in the OP

Please stop, delete, then recreate your container - and if you still have the same issue, please reopen a new issue thread for support.

**Comment by rdwebdesign** [comment]
Try to remove the ENV var from your config:
```
  "PIHOLE_DOCKER_TAG=2021.12",
```

And recreate your container.

**Comment by ogheinz77** [comment]
> What are you using to manage your container? You need to destroy and recreate the container between updates (per the upgrade guide in the readme) else things will start to go wrong. But it is definitely a different issue than that which was reported in the OP
> 
> Please stop, delete, then recreate your container - and if you still have the same issue, please reopen a new issue thread for support.

@PromoFaux, My Pihole runs on a Synology and i am using Synology Docker Manager. When i upgrade Pihole i reset the
                       Container, download the new image and start the Container again. So I don't need to change the settings every
                       time, e.g. Volume and environment settings. When i destroy and recreate the container all these settings are
                       gone, correct?

**Comment by PromoFaux** [comment]
Follow this: (no experience with using synology docker manager. I have a syno, but I just use docker-compose on the command line, this is from a google search)

https://www.wundertech.net/how-to-update-docker-containers-on-a-synology-nas/

There are also a load of stale environment variables in your setup from where you've updated incorrectly in the past. Remove the following from your configuration before starting the new container:

```
PIHOLE_DOCKER_TAG
CORE_VERSION
WEB_VERSION
FTL_VERSION
PIHOLE_VERSION
PIHOLE_ARCH
PIHOLE_TAG
```

Follow that guide above.  Also see here for notes on upgrading : https://github.com/pi-hole/docker-pi-hole#upgrading--reconfiguring

I'm going to ask that you stop replying on this thread about your issue - as it is unrelated to the OP's issue (third  time of me mentioning). If you are still having trouble, please open a _new issue thread_

**Comment by PeterDaveHello** [comment]
Still needs a fix though.

**Comment by thomasmerz** [comment]
This Issue is fixed by [Release 2022.09](https://github.com/pi-hole/docker-pi-hole/releases/tag/2022.09) - so this issue should be closed… @yubiuser 

From my `docker logs pihole`:
```
  Current Pi-hole version is v5.12
  Current AdminLTE version is v5.14
  Current FTL version is v5.17
  Container tag is: 2022.09
```
